plugin/documents.py

```diff
@@ -842,31 +846,37 @@ def _register_async(self) -> None:
     def _on_view_updated_async(self) -> None:
         self._code_lenses_debouncer_async.debounce(
             self._do_code_lenses_async, timeout_ms=self.code_lenses_debounce_time)
-        different, current_region = self._update_stored_region_async()
-        if not different:
+        changed_region, any_different = self._update_stored_selection_async()
+        if not changed_region:
             return
         self._clear_highlight_regions()
         if userprefs().document_highlight_style:
             self._when_selection_remains_stable_async(
-                self._do_highlights_async, current_region, after_ms=self.highlights_debounce_time)
+                self._do_highlights_async, changed_region, after_ms=self.highlights_debounce_time)
         self.do_signature_help_async(manual=False)
 
-    def _update_stored_region_async(self) -> Tuple[bool, sublime.Region]:
+    def _update_stored_selection_async(self) -> Tuple[Optional[sublime.Region], bool]:
```

Comment:
Each call of this function now makes up to three API calls of `View.sel()` (one in `first_selection_region` and you do `list(self.view.sel())` in the code below twice).
It shouldn't be hard to reduce this to a single API call.

Reply:
Good catch! Did not realize (or forgot) that empty regions evaluate to false.